frontend/src/lib/components/ConfirmUpgradeModal/ConfirmUpgradeModal.tsx

```diff
@@ -2,29 +2,25 @@ import { IconCheckCircle } from '@posthog/icons'
 import { LemonButton, LemonModal, Tooltip } from '@posthog/lemon-ui'
 import { useActions, useValues } from 'kea'
 import { useMemo } from 'react'
+import { getProration } from 'scenes/billing/billing-utils'
 import { billingLogic } from 'scenes/billing/billingLogic'
 
 import { confirmUpgradeModalLogic } from './confirmUpgradeModalLogic'
 
 export function ConfirmUpgradeModal(): JSX.Element {
     const { upgradePlan } = useValues(confirmUpgradeModalLogic)
-    const { daysRemaining, daysTotal, billing } = useValues(billingLogic)
+    const { timeRemaining, timeTotal, billing } = useValues(billingLogic)
     const { hideConfirmUpgradeModal, confirm, cancel } = useActions(confirmUpgradeModalLogic)
 
-    const prorationAmount = useMemo(
+    const { prorationAmount, isProrated } = useMemo(
         () =>
-            upgradePlan?.unit_amount_usd
-                ? (parseInt(upgradePlan?.unit_amount_usd) * ((daysRemaining || 1) / (daysTotal || 1))).toFixed(2)
-                : 0,
-        [upgradePlan, daysRemaining, daysTotal]
-    )
-
-    const isProrated = useMemo(
-        () =>
-            billing?.has_active_subscription && upgradePlan?.unit_amount_usd
-                ? prorationAmount !== parseInt(upgradePlan?.unit_amount_usd || '')
-                : false,
-        [billing?.has_active_subscription, prorationAmount]
+            getProration({
```

Comment:
Stripe prorates by the second. If someone leaves the tab open for a long time or simply navigates away for a while without refreshing, this number will also be wrong. I'm not opposed to making this improvement, but I wonder if we should simply tell the user the pro-ration is an overestimate (? It seems like this will always be over) of the actual charge. 

If users want to see the actual amount Stripe will charge, we can pull it from Stripe and display it on the product card after activation.

Reply:
Yeah, we're not going to get this perfect without 1. pulling the value from an upcoming invoice preview and 2. passing in a specific time to prorate to. Both are optimizations that IMO are not worth it right now.

Good call on the `Stripe prorates by the second`, I can update the logic to follow that. 

To your point, I think also adding a `~` or `estimated`  to the number could suffice to account for any slippage in the proration amount.